starport/cmd/validation.go

```diff
@@ -0,0 +1,28 @@
+package starportcmd
+
+import (
+	"context"
+
+	"github.com/spf13/cobra"
+	"github.com/tendermint/starport/starport/pkg/placeholder"
+)
+
+// FlagNoValidation disables validation.
+const FlagNoValidation = "no-validation"
```

Comment:
Do you think there is a use case where we'll want to disable placeholder check?

Reply:
what if placeholders were removed intentionally? and now I want to generate only new stuff?

Reply:
still couldn't get the edge case here. can you please demonstrate me a real world scnerio that can utilize this flag with the `type` or `module import` commands?

in my understanding, if a certain scaffolding related starport command triggers a placeholder replacement in the templates package, this means that this placeholder is actually must be exists in the source code and therefore checking the existence of this placeholder cannot be opt-out.

Reply:
the `module create` is the easiest one. it has placeholders in app.go, if all of them are removed the module is still valid but not registered.

Reply:
I don't think there is any downside of having this feature. It can be used by someone that doesn't want the command to be too verbose, to have too much side effects like this user: https://forum.cosmos.network/t/starport-type-is-too-verbose/4813

Someone can be interested to have the code scaffolded by the `type` command but doesn't want Starport to have any side effect on the current codebase of their app

Reply:
> the module create is the easiest one. it has placeholders in app.go, if all of them are removed the module is still valid but not registered.

In such case we should report to user whether we could made a change in app.go or just created the module and user needs to the writing manually.


> Someone can be interested to have the code scaffolded by the type command but doesn't want Starport to have any side effect on the current codebase of their app

Also in the case, template package already knows if user skipped scaffolding some features and the code that will be generated no longer depends on some certain placeholders.

In both cases, template logic always knows if a placeholder is required and not and in this case there is no need to intorduce another flag to give control on this to user. In-fact this could lead to bugs if user uses the flag in a wrong way.

Reply:
i will remove no-validation from this change. let's tackle no-modification separately

Reply:
So yes, currently there may be the case where, for some commands, the generated code is dependant on changes performed on the existing files through placeholders.

The `--no-modification` flag would require to ensure that all generated code is independent from the placeholders. So this is out of scope for this PR